app/src/main/java/com/android/sample/ui/activity/ActivityForm.kt

```diff
@@ -194,44 +218,56 @@ fun ActivityForm(
             Modifier.padding(end = STANDARD_PADDING.dp)
                 .align(Alignment.CenterVertically)
                 .testTag("iconEndTimeCreate"))
-    if (durationIsSet) Text("Finishing Time: ${duration} (click to change)")
-    else Text("Select End Time")
-  }
-  if (durationIsOpen) {
-    MyTimePicker(
-        onTimeSelected = onSelectDuration,
-        isOpen = durationIsOpen,
-        onCloseRequest = onCloseDuration)
+    if (durationIsSet) Text("Duration Time: $duration (click to change)")
+    else Text("Select Duration")
   }
+
   Spacer(modifier = Modifier.height(STANDARD_PADDING.dp))
 
-  OutlinedTextField(
+  TextFieldWithErrorState(
       value = price,
       onValueChange = onPriceChange,
-      label = { Text("Price") },
-      modifier = Modifier.padding(STANDARD_PADDING.dp).fillMaxWidth().testTag("inputPriceCreate"),
-      placeholder = { Text(text = stringResource(id = R.string.request_price_activity)) },
-      singleLine = true,
-  )
+      label = stringResource(id = R.string.request_price_activity),
+      modifier = Modifier.padding(STANDARD_PADDING.dp).fillMaxWidth(),
+      validation = { price ->
+        when {
```

Comment:
The checks are good here but we should also check that the price is positive. This was initially part of another task but I guess it's ok to add it here along the other checks!

Reply:
It's not a check just displays as an error but doesn't stop it from negativ or NaN

Reply:
Oh I see... Then this will be implemented in another PR, part of my tasks.